woocommerce/woocommerce-admin
# Try ExPlat: Add A/A test

Fixes https://github.com/woocommerce/woocommerce-admin/issues/6660

Adds ExPlat package for use in WooCommerce environments. Also added is a simple A/A test to run as a trial experiment.

Bonus addition is changes to the build files so that changes to `*.ts` files trigger a rebuild on `npm start`

## Test instructions

Its hard to tell if the process is working because the details are in [Calypso's explat package](https://github.com/Automattic/wp-calypso/tree/trunk/packages/explat-client-react-helpers), but inspecting the Network and Console tabs can give us some clues.

1. Make sure you are opted into data sharing
2. Visit the Homescreen
3. See a request in the Network tab to `https://public-api.wordpress.com/wpcom/v2/experiments/0.1.0/assignments/woocommerce`
4. The payload will include an experiment name and anonymous id
5. Turn off data sharing WooCommerce > Settings > Advanced > woocommerc.com
6. Visit the Homescreen
7. See an error in the console about ExPlat turned off due to data sharing. This error will be caught and handled in production and removes the user from the experiment while rendering the default experience

## A Note about IE 11

`window.fetch` does not exist in IE 11, for which we will soon be dropping support soon. Since errors will be thrown in attempts to call `fetch` which will remove the user from the experiment, I've decided to simply let that happen instead of explicitly handling that scenario with a check for fetch and manually throwing an Error.
